git/setup.rst

```diff
@@ -111,18 +111,44 @@ To have your credentials cached for 1 hour (3600 seconds):
 
 **Linux users can alternatively** have their `credentials stored on disk <http://git-scm.com/docs/git-credential-store>`_ in a :file:`~/.git-credentials` file.
 Only do this for machines where you can ensure some level of security.
+To do this securely on USDF, set the credential file to be only read/write by your user: ``chmod 0600 ~/.git-credentials``.
 
 .. code-block:: bash
 
    git config --global credential.helper store
 
 Once a credential helper is enabled, the next time you ``git push``, you will add your credentials to the helper.
 
-Remember that if you have 2FA enabled, you will create and use a `personal access token <https://github.com/settings/tokens>`_ instead of your GitHub password.
+If you are using HTTPS access, you will need to create and use a :ref:`personal access token <git-auth-tokens>` instead of your GitHub password, as described below.
 
-If you find that ``git push`` is not working but also not asking you for credentials, you may need to manually insert the username/password or token into the credential store or macOS keychain.
+.. _git-auth-tokens:
 
-The DM Git LFS documentation has further information about :ref:`authenticating with our LFS storage backend <git-lfs-auth>`.
+Git authentication tokens
+-------------------------
+
+Due to GitHub's authentication interface, you must use a personal access token instead of a password when pushing to GitHub via HTTPS.
+GitHub access via ssh does not require a token, but :doc:`Git LFS repos <git-lfs>` work best via HTTPS.
+A credential helper, as :ref:`described above <git-credential-helper>`, will let you cache this token, so you do not have to save it (you can only view it on GitHub when you generate it).
+Create a personal token at `GitHub Personal Access Tokens <https://github.com/settings/tokens>`_, using the "generate new token" button in the upper right, with the "classic" interface.
+Give it a name (e.g. "LSST USDF"), choose an expiration date (when the token expires you will have to generate a new one and re-enter it into your credential cache), and set the following permissions::
+
+  repo->public_repo
+  admin:org->read:org
```

Comment:
This is misleading.  `admin:org` should not be checked.  `repo` may or may not need to be checked; mostly we avoid private repos, but there are a few.